rustls-post-quantum/src/hybrid.rs

```diff
@@ -0,0 +1,178 @@
+use rustls::crypto::{ActiveKeyExchange, CompletedKeyExchange, SharedSecret, SupportedKxGroup};
+use rustls::ffdhe_groups::FfdheGroup;
+use rustls::{Error, NamedGroup, ProtocolVersion};
+
+use crate::INVALID_KEY_SHARE;
+
+/// A generalization of hybrid key exchange.
+#[derive(Debug)]
+pub(crate) struct Hybrid {
+    pub(crate) classical: &'static dyn SupportedKxGroup,
+    pub(crate) post_quantum: &'static dyn SupportedKxGroup,
+    pub(crate) name: NamedGroup,
+    pub(crate) layout: Layout,
+}
+
+impl SupportedKxGroup for Hybrid {
+    fn start(&self) -> Result<Box<dyn ActiveKeyExchange>, Error> {
+        let classical = self.classical.start()?;
+        let post_quantum = self.post_quantum.start()?;
+
+        let combined_pub_key = self
+            .layout
+            .concat(post_quantum.pub_key(), classical.pub_key());
+
+        Ok(Box::new(ActiveHybrid {
+            classical,
+            post_quantum,
+            name: self.name,
+            layout: self.layout,
+            combined_pub_key,
+        }))
+    }
+
+    fn start_and_complete(&self, client_share: &[u8]) -> Result<CompletedKeyExchange, Error> {
+        let Some((post_quantum_share, classical_share)) = self
+            .layout
+            .split_received_client_share(client_share)
+        else {
+            return Err(INVALID_KEY_SHARE);
+        };
```

Comment:
Because I don't like the multi-line scrutinee in the let-else, suggest using `ok_or(INVALID_KEY_SHARE)?` instead (also seems generally simpler).